File: 2017/Case.FreeBenchmarking/Case.FreeBenchmarking/Classes/Entry/cmd.vb
Imports Autodesk.Revit.Attributes
Imports Autodesk.Revit.DB
Imports Autodesk.Revit.UI

<Transaction(TransactionMode.Manual)>
Public Class cmd

  Implements IExternalCommand

  ''' <summary>
  ''' Command Entry Point
  ''' </summary>
  ''' <param name="commandData"></param>
  ''' <param name="message"></param>
  ''' <param name="elements"></param>
  ''' <returns></returns>
  ''' <remarks></remarks>
  Public Function Execute(ByVal commandData As ExternalCommandData,
                          ByRef message As String,
                          ByVal elements As ElementSet) As Result Implements IExternalCommand.Execute

    Try

      ' Version
      If Not commandData.Application.Application.VersionName.Contains("2014") Then

        ' Failure
        message = "This Add-In was built for Revit 2014, please contact CASE for assistance..."
        Return Result.Failed

      End If

      ' Construct and Display the Form
      Using m_d As New form_Main(New clsSettings(commandData, elements))
        m_d.ShowDialog()
      End Using

      ' Success
      Return Result.Succeeded

    Catch ex As Exception

      ' Failure
      message = ex.Message
      Return Result.Failed

    End Try

  End Function

End Class

File: 2018/Case.ViewCreator/Case.ViewCreator/Classes/Entry/cmd.cs
using System;

using Autodesk.Revit.Attributes;
using Autodesk.Revit.DB;
using Autodesk.Revit.UI;

namespace Case.ViewCreator
{

  /// <summary>
  /// Revit 2018 Command Class
  /// </summary>
  /// <remarks></remarks>
  [Transaction(TransactionMode.Manual)]
  public class cmd : IExternalCommand
  {

    /// <summary>
    /// Command Entry Point
    /// </summary>
    /// <param name="commandData">Input argument providing access to the Revit application and documents</param>
    /// <param name="message">Return message to the user in case of error or cancel</param>
    /// <param name="elements">Return argument to highlight elements on the graphics screen if Result is not Succeeded.</param>
    /// <returns>Cancelled, Failed or Succeeded</returns>
    public Result Execute(ExternalCommandData commandData,
                            ref string message,
                            ElementSet elements)
    {
      try
      {

        // Version
        if (!commandData.Application.Application.VersionName.Contains("2018"))
        {
          message = "This Add-In was built for Revit 2018, please contact CASE for assistance...";
          return Result.Failed;
        }

        // Construct and Display the form
        form_Main frm = new form_Main(commandData);
        frm.ShowDialog();

        // Return Success
        return Result.Succeeded;
      }
      catch (Exception ex)
      {

        // Failure Message
        message = ex.Message;
        return Result.Failed;

      }
    }
  }
}